pkg/cassandra/datacenter.go

```diff
@@ -88,9 +93,41 @@ func NewDatacenter(klusterKey types.NamespacedName, template *DatacenterConfig)
 		dc.Spec.Resources = *template.Resources
 	}
 
+	if len(template.MgmtAPIHeap) != 0 {
+		if err := SetMgmtAPIHeap(dc, template.MgmtAPIHeap); err != nil {
+			return nil, err
+		}
+	}
+
 	return dc, nil
 }
 
+// SetMgmtAPIHeap sets the management API heap size on a CassandraDatacenter
+func SetMgmtAPIHeap(dc *cassdcapi.CassandraDatacenter, heapSize string) error {
+	// TODO: it would be nice to have a generic `StrategicMergePatch` method which produces merged API objects instead
```

Comment:
If we were exposing a PodTemplateSpec with k8ssandra-operator and passing it through then I could see using a generic merge approach, but we're not doing that. In fact I would ideally prefer to minimize the use the PodTemplateSpec in the CassandraDatacenter to the extent we can.

Reply:
Done.